src/api.cpp

```diff
@@ -5168,11 +5265,27 @@ cl_int CLVK_API_CALL clEnqueueFillImage(
     // TODO use Vulkan clear commands when possible
     // TODO use a shader when better
 
+    auto img = static_cast<cvk_image*>(image);
+
+    // Fill
+    cvk_image::fill_pattern_array pattern;
+    size_t pattern_size;
+    img->prepare_fill_pattern(fill_color, pattern, &pattern_size);
+
+    if (img->is_backed_by_buffer_view()) {
+        auto cmd = new cvk_command_fill_buffer(
```

Comment:
This will expose the wrong command type to applications.

Reply:
Fixed